App/Services/MutagenController.cs

```diff
@@ -534,25 +538,44 @@ private void StartDaemonProcess()
         Directory.CreateDirectory(_mutagenDataDirectory);
         var logPath = Path.Combine(_mutagenDataDirectory, "daemon.log");
         var logStream = new StreamWriter(logPath, true);
+        try
+        {
+            _daemonProcess = new Process();
+            _daemonProcess.StartInfo.FileName = _mutagenExecutablePath;
+            _daemonProcess.StartInfo.Arguments = "daemon run";
+            _daemonProcess.StartInfo.Environment.Add("MUTAGEN_DATA_DIRECTORY", _mutagenDataDirectory);
+            // hide the console window
+            _daemonProcess.StartInfo.CreateNoWindow = true;
+            // shell needs to be disabled since we set the environment
+            // https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.environment?view=net-8.0
+            _daemonProcess.StartInfo.UseShellExecute = false;
+            _daemonProcess.StartInfo.RedirectStandardError = true;
+            // TODO: log exited process
+            // _daemonProcess.Exited += ...
+            if (!_daemonProcess.Start())
+                throw new InvalidOperationException("Failed to start mutagen daemon process, Start returned false");
+
+            var writer = new LogWriter(_daemonProcess.StandardError, logStream);
+            Task.Run(() => { _ = writer.Run(); });
+            _logWriter = writer;
+        }
+        catch
+        {
+            try
+            {
+                _daemonProcess?.Kill();
+            }
+            catch
+            {
+                // ignored
+            }
 
-        _daemonProcess = new Process();
-        _daemonProcess.StartInfo.FileName = _mutagenExecutablePath;
-        _daemonProcess.StartInfo.Arguments = "daemon run";
-        _daemonProcess.StartInfo.Environment.Add("MUTAGEN_DATA_DIRECTORY", _mutagenDataDirectory);
-        // hide the console window
-        _daemonProcess.StartInfo.CreateNoWindow = true;
-        // shell needs to be disabled since we set the environment
-        // https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.environment?view=net-8.0
-        _daemonProcess.StartInfo.UseShellExecute = false;
-        _daemonProcess.StartInfo.RedirectStandardError = true;
-        // TODO: log exited process
-        // _daemonProcess.Exited += ...
-        if (!_daemonProcess.Start())
-            throw new InvalidOperationException("Failed to start mutagen daemon process, Start returned false");
-
-        var writer = new LogWriter(_daemonProcess.StandardError, logStream);
-        Task.Run(() => { _ = writer.Run(); });
-        _logWriter = writer;
+            _daemonProcess?.Dispose();
+            _logWriter?.Dispose();
+            _daemonProcess = null;
+            _logWriter = null;
```

Comment:
If we throw an exception starting the daemon, the `logStream` never gets turned into `_logWriter` and will leave open the logfile, which makes retries fail.